paasta_tools/cli/cmds/status.py

```diff
@@ -1878,6 +1882,250 @@ def print_kafka_status(
     return 0
 
 
+class EtcdLockServerStatus(TypedDict, total=False):
+    observedGeneration: int
+    available: str
+    clientServiceName: str
+
+
+class LockServerStatus(TypedDict, total=False):
+    etcd: EtcdLockServerStatus
+
+
+class VitessClusterCellStatus(TypedDict, total=False):
+    pendingChanges: str
+    gatewayAvailable: str
+
+
+class VitessClusterKeyspaceStatus(TypedDict, total=False):
+    pendingChanges: str
+    cells: List[str]
+    desiredShards: int
+    shards: int
+    readyShards: int
+    updatedShards: int
+    desiredTablets: int
+    tablets: int
+    readyTablets: int
+    updatedTablets: int
+
+
+class VitessDashboardStatus(TypedDict, total=False):
+    available: str
+    serviceName: str
+
+
+class VTAdminStatus(TypedDict, total=False):
+    available: str
+    serviceName: str
+
+
+class OrphanStatus(TypedDict, total=False):
+    reason: str
+    message: str
+
+
+class VitessClusterStatus(TypedDict, total=False):
+    observedGeneration: int
+    globalLockserver: LockServerStatus
+    gatewayServiceName: str
+    vitessDashboard: VitessDashboardStatus
+    cells: Dict[str, VitessClusterCellStatus]
+    keyspaces: Dict[str, VitessClusterKeyspaceStatus]
+    vtadmin: VTAdminStatus
+    orphanedCells: Dict[str, OrphanStatus]
+    orphanedKeyspaces: Dict[str, OrphanStatus]
+
+
+def print_vitess_status(
```

Comment:
might be a useful refactor in the future to break each of the sections into their own function so that `print_vitess_status()` isn't so long